src/go/k8s/apis/redpanda/v1alpha1/console_enterprise_types.go

```diff
@@ -71,3 +74,47 @@ type EnterpriseLoginGoogleDirectory struct {
 	// TargetPrincipal is the user that shall be impersonated by the service account
 	TargetPrincipal string `json:"targetPrincipal"`
 }
+
+// CloudConfig contains configurations for Redpanda cloud. If you're running a
+// self-hosted installation, you can ignore this
+type CloudConfig struct {
+	PrometheusEndpoint *PrometheusEndpointConfig `json:"prometheusEndpoint"`
+}
+
+// PrometheusEndpointConfig configures the Prometheus endpoint that shall be
+// exposed in Redpanda Cloud so that users can scrape this URL to
+// collect their dataplane's metrics in their own time-series database.
+type PrometheusEndpointConfig struct {
+	Enabled   bool              `json:"enabled"`
+	BasicAuth BasicAuthConfig   `json:"basicAuth"`
+	Config    *PrometheusConfig `json:"prometheus"`
+}
+
+// BasicAuthConfig are credentials that will be required by the user in order to
+// scrape the endpoint
+type BasicAuthConfig struct {
+	Username    string                 `json:"username"`
+	PasswordRef corev1.ObjectReference `json:"passwordRef"`
+}
+
+// PrometheusConfig is configuration of prometheus instance
+type PrometheusConfig struct {
+	// Address to Prometheus endpoint
+	Address string `json:"address"`
```

Comment:
Learning q: Is this endpoint different from the one we install [here](https://github.com/redpanda-data/cloudv2/blob/1fe491da55fb5e9aac48038a35e4544997235ee3/terraform/provisioners/kubernetes-redpanda-eks/terraform-kubernetes-redpanda-eks/main.tf#L86)?

Reply:
I don't know 🤔

Reply:
I'm just wondering if this should not be given by the user but inferred by the operator

Reply:
oh I see what you mean, but how would you infer it? By using some default port and address? e.g. the namespace is definitely not standard one. I think we should just pass it in explicitly